bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/VoiceHelper.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.voice.internal;
+
+import java.util.Locale;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.voice.TTSService;
+import org.eclipse.smarthome.core.voice.Voice;
+
+/**
+ * {@link VoiceHelper} defines a utility method to consume sorted voices.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@NonNullByDefault
+public interface VoiceHelper {
```

Comment:
> What is the issue, if the getAllVoices method returns a sorted set?

Is it common practice in other ESH interfaces that sorted results are returned by `getAll..()` methods? It may add unnecessary sorting overhead when sorting it not required or when the result is sorted in another way. Though this will probably not be a real issue because of the small set sizes.

Reply:
Not necessarily common practice, but where useful and not any relevant overhead, it is imho fully ok to return a SortedSet to make life easier. Especially with Sets, it can be pretty frustrating if HashSets are used and the order in the UI is differently every day...

Reply:
I can change the code so `getAllVoices()` returns a sorted set based on the locale of the LocaleProvider. Should the sorted characteristic of the set also be documented in the interface JavaDoc? If this is kept an internal implementation detail the API will also not be broken when this is changed in the future.

Reply:
If we consider it a feature, please also document it in the JavaDoc and add a test to `VoiceManagerTest`.

Reply:
The code has been updated although adding a test for the sorting is a bit more work because the stubs don't allow for much customization. What would be the best solution for that? Adapting stubs for more customization or using Mockito instead?